File: solutions/scala/068-task-scheduler.scala
/**
 * Problem 68: Task Scheduler (LeetCode 621)
 * Difficulty: Med
 * Language: Scala
 */
def leastInterval(tasks: Array[Char], n: Int): Int = {
  val freqs = tasks.groupBy(identity).values.map(_.length).toList
  val maxFreq = freqs.max
  val maxCount = freqs.count(_ == maxFreq)
  tasks.length max ((maxFreq - 1) * (n + 1) + maxCount)
}

File: solutions/elixir/068-task-scheduler.ex
# Problem 68: Task Scheduler (LeetCode 621)
# Difficulty: Med
# Language: Elixir
# 
defmodule TaskScheduler do
  def least_interval(tasks, n) do
    freqs = tasks |> Enum.frequencies() |> Map.values()
    max_freq = Enum.max(freqs)
    max_count = Enum.count(freqs, &(&1 == max_freq))
    max(length(tasks), (max_freq - 1) * (n + 1) + max_count)
  end
end
